### PR TITLE
Add docs templates

### PR DESCRIPTION
**Pull request by janetkuo**
Published in http://janetkuo.github.io/docs/templates/

@kubernetes/docs

### CODE REVIEW

**Comment by erictune** [comment]
:+1:

File: docs/templates/how-tos.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+## Before you begin
+Prerequisites (we can assume that users have completed quickstart)
+
+## Doing Task 1
+If there are multiple ways to complete a task, use a tab widget to describe it.
+
+## Doing Task 2
+
+## Doing Task N
```
Comment (bgrant0607) [inline docs/templates/how-tos.md]: If we're planning to favor smaller, more focused docs, I'm not sure we want to encourage docs with N tasks.

File: docs/templates/index.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+---
+
+When writing docs, start with one of the five templates we provide:
```
Comment (bgrant0607) [inline docs/templates/index.md]: Do we need to put these docs in any special path so they appear prominently on the site?

File: docs/templates/how-tos.md (hunk 1 above)
Comment (pwittrock) [inline docs/templates/how-tos.md]: Should we put an upper limit on N?  5?

**Comment by pwittrock** [comment]
The tutorial and the how-to seem pretty similar.  What are the key differences?

**Comment by janetkuo** [comment]
> The tutorial and the how-to seem pretty similar. What are the key differences?

@pwittrock as described in http://janetkuo.github.io/docs/templates/, the key difference is that:
- **Tutorial:** step-by-step guide (in sequence) showing how things work; e.g., [updating live pods](http://kubernetes.io/docs/user-guide/update-demo/)
- **How-tos:** task-based, with multiple related but independent tasks (can be executed in random order); e.g., [managing resources](http://kubernetes.io/docs/user-guide/managing-deployments/). 

If it's too confusing we can change the name of the template.

File: docs/templates/how-tos.md (hunk 1 above)
Comment (janetkuo) [inline docs/templates/how-tos.md]: N <= 5 sounds good.

File: docs/templates/index.md (hunk 2 above)
Comment (janetkuo) [inline docs/templates/index.md]: That'd be good for people wanting to contribute the docs. Probably under "Writing Guides" or "Writing Documents".

**Comment by johndmulhausen** [comment]
This is not how we will want to do templates; we should be using Jekyll layouts for this. It's fine to have documents that generally discuss what the aim of certain templates are, but this is not an acceptable way to enforce a page type system because it isn't strong enough nor very convenient to use. If our page types were layouts, we could just go:

```
---
layout: tutorial
---
```

At the top of every page that follows a type, then use the layout's defaults to instruct the user on how to fill out the page. I'm going to leave this PR open for reference but code up something that is more robust as a replacement.

All this said, it's a good capturing of what we want to do with templates and I appreciate everybody stepping in while I had the flu.

**Comment by bgrant0607** [comment]
This needs to be aligned with the overall proposal, which also needs review:
https://docs.google.com/document/d/12Nfj2E8tZL-CMnnpkSH9m5k_oqd4qJh4qDwYDxPGJJI/edit?ts=5705b1fc#

We're also going to prototype the proposal's nav structure and a couple concepts/APIs.

We can revisit these templates afterwards.

File: docs/templates/index.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+Template | When to Use This | Download Link
+--- | --- | --- | ---
+[Overview](/docs/templates/overview/) | When you want to write an overview of X and its most commonly used elements (e.g. overview of APIs) | [download this](https://github.com/kubernetes/kubernetes.github.io/tree/master/docs/templates/overview.md)
+[Concept or in-depth guide](/docs/templates/concept-guide/) | When you want to explain X and how it works in detail (e.g. Deployment object concept) | [download this](https://github.com/kubernetes/kubernetes.github.io/tree/master/docs/templates/concept-guide.md)
+[Quickstart](/docs/templates/quickstart/) | When you want to get users working with the technology quickly by completing an introductory task within 5 minutes | [download this](https://github.com/kubernetes/kubernetes.github.io/tree/master/docs/templates/quickstart.md)
```
Comment (bgrant0607) [inline docs/templates/index.md]: I think we should eliminate "Quickstart". I don't understand the difference between it and the other types of docs.

File: docs/templates/how-tos.md (hunk 1 above)
Comment (bgrant0607) [inline docs/templates/how-tos.md]: N=1?

**Comment by bgrant0607** [comment]
Mocks:
https://docs.google.com/presentation/d/14mJwJGlwBJHIpMRGR-nYMSyyNk-kRpabyz74-YYXeTo/edit#slide=id.g12d21f53ed_2_11

**Comment by bgrant0607** [comment]
Closing now to avoid confusion, since these don't match the mocks.

**Pull request closed without merging**